source/Interfaces/MainInterface.cpp

```diff
@@ -0,0 +1,297 @@
+                                                                               "Maze Window") {
+        menuBtn.setButtonSize({200,50});
+        menuBtn.setBackColor(sf::Color::Black);
+        menuBtn.setMFont();
+        menuBtn.setString("Menu");
+        menuBtn.setTextColor(sf::Color::White);
+        menuBtn.setPosition({0,0});
+
+        inventoryBTN.setButtonSize({200,50});
+        inventoryBTN.setBackColor(sf::Color::Black);
+        inventoryBTN.setMFont();
+        inventoryBTN.setString("Inventory");
+        inventoryBTN.setTextColor(sf::Color::White);
+        inventoryBTN.setPosition({200,0});
+
+        exitBtn.setButtonSize({200,50});
+        exitBtn.setBackColor(sf::Color::Black);
+        exitBtn.setMFont();
+        exitBtn.setString("Exit");
+        exitBtn.setTextColor(sf::Color::White);
+        exitBtn.setPosition({400,0});
+    }
+
+    /**
+     * @brief Creates a vector representation of the maze grid.
+     *
+     * @param grid         The WorldGrid representing the maze.
+     * @param type_options The type options for symbols.
+     * @param item_set     The set of items in the maze.
+     * @param agent_set    The set of agents in the maze.
+     *
+     * @return A vector of strings representing the maze grid.
+     */
+    std::vector<std::string> MainInterface::CreateVectorMaze(const WorldGrid &grid, const type_options_t &type_options,
+                                                             const item_set_t &item_set, const agent_set_t &agent_set) {
+        std::vector<std::string> symbol_grid(grid.GetHeight());
+        // Load the world into the symbol_grid;
+        for (size_t y = 0; y < grid.GetHeight(); ++y) {
+            symbol_grid[y].resize(grid.GetWidth());
+            for (size_t x = 0; x < grid.GetWidth(); ++x) {
+                symbol_grid[y][x] = type_options[grid.At(x, y)].symbol;
+            }
+        }
+
+        // Add in the agents / entities
+        for (const auto &entity_ptr: item_set) {
+            GridPosition pos = entity_ptr->GetPosition();
+            symbol_grid[pos.CellY()][pos.CellX()] = '+';
+        }
+
+        for (const auto &agent_ptr: agent_set) {
+            GridPosition pos = agent_ptr->GetPosition();
+            char c = '*';
+            if (agent_ptr->HasProperty("char")) {
+                c = static_cast<char>(agent_ptr->GetProperty("char"));
+            }
+            symbol_grid[pos.CellY()][pos.CellX()] = c;
+        }
+        return symbol_grid;
+    }
+
+    /**
+    * @brief Draws the maze grid and entities on the SFML window.
+    *
+    * @param grid         The WorldGrid representing the maze.
+    * @param type_options The type options for symbols.
+    * @param item_set     The set of items in the maze.
+    * @param agent_set    The set of agents in the maze.
+    */
+    void MainInterface::DrawGrid(const WorldGrid &grid, const type_options_t &type_options,
```

Comment:
This function is getting a bit long -- you may want to look into a good way to split it up.  One simple refactor might be to pull out the drawing of a single cell into its own `DrawCell` function (where you pass in the coordinates and any other info it might need.)  Even within `DrawCell`, you may want the switch to call sub functions like `DrawCell_Wall`, `DrawCell_Empty`, etc...  It will make reading the function much easier.